routing/result_interpretation.go

```diff
@@ -349,8 +362,7 @@ func (i *interpretedResult) processPaymentOutcomeIntermediate(
 	// control.
 	case *lnwire.FailAmountBelowMinimum,
 		*lnwire.FailFeeInsufficient,
-		*lnwire.FailIncorrectCltvExpiry,
-		*lnwire.FailChannelDisabled:
```

Comment:
Yeah I think the root cause here is an incorrect mental model that assumed that the channel disabled error would only ever be sent by the final hop.

Reply:
Have you confirmed that the second chance logic _won't_ apply here? Or is it that it does, but we end up penalizing an intermediate hop which ends up messing up later path finding attempts?

Reply:
Looks like we're lacking a test case involving `lnwire.FailChannelDisabled` here all together. `lnd` doesn't _yet_ actually send out this error ([to be fixed eventually by this PR](https://github.com/lightningnetwork/lnd/pull/5565), so that likely influenced this logic when it was originally being written.

Reply:
It applies, but the second chance is very likely to fail again if the problem is the mobile device being temporarly offline. However, a third retry is rejected at path finding since the intermediate hop is penalized (On the 1st attempt & then on 2nd chance attempt too).